### PR TITLE
Added installation option via third-party installer n-install

### PR DESCRIPTION
**Pull request by mklement0**
as discussed in [#286](https://github.com/tj/n/issues/286#issuecomment-122860542).

Also streamlined description of the manual installation process.

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -11,15 +11,21 @@ Simple flavour of node/iojs binary management, no subshells, no profile setup, n
...
     $ npm install -g n
 
-or
+or, by cloning this repo and running:
 
-    $ make install
+    $ PREFIX=$HOME make install
 
-to `$HOME`. Prefix later calls to `n` with `N_PREFIX=$HOME`
+to install to `$HOME`. Prefix later calls to `n` with `N_PREFIX=$HOME` or export `N_PREFIX` in your shell initialization file.
```
Comment (maxrimue) [inline Readme.md]: Shouldn't we instead suggest installing n with just `$ make install`, but note that you can adjust the path for installation? Also, when installing n while passing the PREFIX variable to the Makefile, the $N_PREFIX won't get adjusted. The PREFIX variable is the path where n will download Node.js/io.js, the $N_PREFIX variable is the path where Node.js/io.js will be installed into.

Comment (mklement0) [inline Readme.md]: Actually, both `PREFIX` and `N_PREFIX` refer to the _same_ directory and default to `/usr/local/bin`: they both refer to the "prefix directory" into which the *active* node version is installed, and in whose `n` *subdirectory*  `n` is expected to be installed.

The fact that they refer to the same dir. is why I suggested eliminating `PREFIX` from the Makefile and using `N_PREFIX` there as well.

Comment (mklement0) [inline Readme.md]: As for using just `make install`: I'm happy to elaborate; I stuck with `$HOME`, because that's what the existing text seemed to describe, albeit inconsistently.

However, note that just `make install` will try to install to `/usr/local`, which will typically require `sudo`.

I think it actually makes sense to suggest a _user-level_ install - either by demonstrating use of `[N_]PREFIX`, as I've suggested, and/or even changing the _default_ location to `$HOME` (which is part of what the original description seemed to imply).

**Comment by maxrimue** [comment]
@mklement0 The `PREFIX` in the Makefile is the path where n is, the `N_PREFIX` is the path where Node.js will be, they're actually different. If you install n with `PREFIX=$HOME make install`, you'll find a bin folder, in which the n executable will be. If you then call this executable and tell it to install the latest Node.js, for example, it will download it into `$N_PREFIX/n/versions/...`, and install it into `$N_PREFIX/`. So, if you install n with a custom `PREFIX`, n itself and Node.js will be installed into different directories, as long as you don't export the `N_PREFIX`, to not use the default `/usr/local/` path, manually.

**Comment by mklement0** [comment]
@maxrimue Oops! You're absolutely right. (Guess I was confused by the fact that the two variables have the same default value, and, if you use my installer and update script, are kept the same behind the scenes).

I'll amend my PR.

**Comment by mklement0** [comment]
@maxrimue Hope it hasn't gotten too wordy, but I wanted to get the `PREFIX`, `N_PREFIX` distinction in there; I've also extended the passage about n-install accordingly, and now also mention the `n-uninstall` and `n-update` scripts.
